monitor.cpp

```diff
@@ -3,36 +3,55 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <vector>
+#include <string>
+
+using std::cout, std::vector, std::pair, std::string, \
+std::flush, std::this_thread::sleep_for, std::chrono::seconds, std::endl;
+
+enum class VitalType {
+  TEMPERATURE,
+  PULSE_RATE,
+  SPO2
+};
+
+vector<pair<float, float>> vitalLimits = {
```

Comment:
Nice approach with the vector! Minimal work to add a parameter to the report!
Suggestion: in this approach, we need to be careful to have the sequence of limits to match the sequence of messages - they are in different portions of the code.
Can make a structure to combine the vital type, range and message (id) into a struct, and keep a vector of that struct.